node/src/main/scala/io/casperlabs/node/api/graphql/schema/globalstate/types/package.scala

```
@@ -0,0 +1,288 @@
+          value = ipc.KeyURef.AccessRights.ADD_WRITE
+        ),
+        EnumValue(
+          "READ_ADD_WRITE",
+          value = ipc.KeyURef.AccessRights.READ_ADD_WRITE
+        )
+      )
+    )
+
+  lazy val KeyAddress =
+    ObjectType(
+      "KeyAddress",
+      fields[Unit, ipc.KeyAddress](
+        Field(
+          "value",
+          StringType,
+          resolve = c => Base16.encode(c.value.account.toByteArray)
+        )
+      )
+    )
+
+  lazy val KeyHash = ObjectType(
+    "KeyHash",
+    fields[Unit, ipc.KeyHash](
+      Field(
+        "value",
+        StringType,
+        resolve = c => Base16.encode(c.value.key.toByteArray)
+      )
+    )
+  )
+
+  lazy val KeyURef = ObjectType(
+    "KeyUref",
+    fields[Unit, ipc.KeyURef](
+      Field(
+        "uref",
+        StringType,
+        resolve = c => Base16.encode(c.value.uref.toByteArray)
+      ),
+      Field(
+        "accessRights",
+        AccessRightsEnum,
+        resolve = _.value.accessRights
+      )
+    )
+  )
+
+  lazy val KeyLocal = ObjectType(
+    "KeyLocal",
+    fields[Unit, ipc.KeyLocal](
+      Field(
+        "seed",
+        StringType,
+        resolve = c => Base16.encode(c.value.seed.toByteArray)
+      ),
+      Field(
+        "keyHash",
+        StringType,
+        resolve = c => Base16.encode(c.value.keyHash.toByteArray)
+      )
+    )
+  )
+
+  lazy val KeyUnion = UnionType(
+    "KeyUnion",
+    types = List(
+      KeyAddress,
+      KeyHash,
+      KeyURef,
+      KeyLocal
+    )
+  )
+
+  lazy val Key = ObjectType(
+    "Key",
+    fields[Unit, ipc.Key](
+      Field(
+        "value",
+        KeyUnion,
+        resolve = _.value.keyInstance match {
+          case ipc.Key.KeyInstance.Account(value) => value
+          case ipc.Key.KeyInstance.Hash(value)    => value
+          case ipc.Key.KeyInstance.Uref(value)    => value
+          case ipc.Key.KeyInstance.Local(value)   => value
+          case ipc.Key.KeyInstance.Empty          => ???
```

I'm not sure if I have to handle this case.

---

@goral09 Is it possible that I get `Empty` here? I didn't find in the `ipc.proto`. Probably autogenerated by protobuf.

---

It's generated, yes. You shouldn't get it and I think it's best to omit that case in the pattern matching to fail fast and fix.